### DISCUSSION TITLE
Best Approach / Examples / Resources - Additional 'dist-build' features?

### DISCUSSION THREAD

**Discussion by wpbuilds** [understrap/understrap-child#352] [Q&A]
Hi there,
               I would love to add a couple of features to the 'npm run dist-build' process (For my own personal distribution scenarios). After creating the 'dist' I often have to manually update a few things before pushing the child-theme to a live environment. 

EG 1: I would like to be able to change the 'Theme Name:' within the styles.css.

EG 2: I would like to be able to change the 'screenshot.png'.

Any approaches/suggestions/resources/examples that could point me in the right direction to achieving any of the above would be much appreciated :-)

**Comment by bacoords** [comment]
Can you explain what you mean here? I'm not sure if there's a good way to automate these other than to update the originals in the theme and they should get copied when you run the `npm run dist-build`

**Reply by wpbuilds** [reply]
Hi @bacoords thanks for responding. Maybe not the best explanation by me... soz. To cut to the chase - I suppose I am interested in adding/creating my own NPM based commands/functions and where a good place to start and learn might be?

**Reply by bacoords** [reply]
Ah- basically you can write your own JavaScript files and then add them to the 'scripts' section of the package.json file. You'll see some of our commands work like that: 

`"dist-build": "node src/build/dist-build.js",` would mean that running `npm run dist-build` will essentially pull up that JavaScript file and execute it.

**Reply by wpbuilds** [reply]
Thanks! I shall dig deeper. Thanks for your time @bacoords
